src/trixi_timeit.jl

```diff
@@ -0,0 +1,71 @@
+# Enable debug timings `@trixi_timeit timer() "name" stuff...`.
+# This allows us to disable timings completely by executing
+# `TimerOutputs.disable_debug_timings(TrixiBase)`
+# and to enable them again by executing
+# `TimerOutputs.enable_debug_timings(TrixiBase)`
+timeit_debug_enabled() = true
+
+"""
+    disable_debug_timings()
+
+Disable all [`@trixi_timeit`](@ref) timings.
+The timings should be optimized away, allowing for truly zero-overhead.
+Enable timings again with [`enable_debug_timings`](@ref).
+
+See also [`enable_debug_timings`](@ref), [`@trixi_timeit`](@ref).
+"""
+disable_debug_timings() = TimerOutputs.disable_debug_timings(@__MODULE__)
+
+"""
+    enable_debug_timings()
+
+Enable all [`@trixi_timeit`](@ref) timings (default behavior).
+
+See also [`disable_debug_timings`](@ref), [`@trixi_timeit`](@ref).
+"""
+enable_debug_timings() = TimerOutputs.enable_debug_timings(@__MODULE__)
+
+# Store main timer for global timing of functions.
+# Always call timer() to hide implementation details.
+const main_timer = TimerOutput()
```

Comment:
@efaulhaber Thanks for tackling this! However, I am wondering whether adding the `main_timer` here is such a good idea in the long run - won't this preclude multiple packages depending on TrixiBase.jl from working together properly, since they now all share the same "main timer"?

If you concur, I think it would be good to create an issue for it such that we do not forget about it. But it may also very well be that I'm just overlooking something here 😅

cc @ranocha

Reply:
You could also argue that this means that this now allows packages to work together properly since they share the same timer. I don't have a good estimate of which perspective is more relevant in the long run.

Reply:
These are good points. Thanks for clarifying; no need to act right now from my POV.